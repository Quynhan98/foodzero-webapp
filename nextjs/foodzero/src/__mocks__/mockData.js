export const AVATAR_MOCK =
  'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fjohn-doe.webp?alt=media&token=6963f744-1ab8-4b09-971d-2486cd5375bf'

export const PRICE_LIST_MOCK = {
  name: 'Deep Sea Snow White Cod Fillet',
  price: 20,
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. ',
}

export const CARD_BLOG = {
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2FFeature%20Image.webp?alt=media&token=766a34ea-32e0-40e7-90e6-299f9af22eb1',
  title: 'Fruit and vegetables and protection against diseases',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
  category: 'Fashion',
  srcAvatar: AVATAR_MOCK,
  name: 'Julie Christie',
  date: '2022-12-06T23:29:35.000Z',
  numberOfComments: 2,
}

export const CARD_FEATURE = {
  title: 'Premium Quality',
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2FCircleIcon.webp?alt=media&token=1df373f3-27a3-45f6-82fa-d997c47c718f',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Neque congue arcu',
}

export const CARD_CATEGORY = {
  href: '/',
  category: 'Starters',
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fcategory.webp?alt=media&token=9f307d7c-b242-479d-943a-4ae672ac601e',
}

export const USERS_MOCK = [
  {
    id: 'gff1',
    email: 'foodzero@gmail.com',
    password: '123456789',
  },
]

export const QUOTE_MOCK = [
  {
    id: '1',
    authorName: 'John Doe',
    job: 'Blogger',
    quote:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Purus lorem id penatibus impedient. Turpis egestas ultricies purus  Lorem ipsum dolor sit amet.',
    avatar:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fjohn-doe.webp?alt=media&token=6963f744-1ab8-4b09-971d-2486cd5375bf',
  },
  {
    id: '2',
    authorName: 'Julie Christie',
    job: 'Blogger',
    quote:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Purus lorem id penatibus impedient. Turpis egestas ultricies purus  Lorem ipsum dolor sit amet1.',
    avatar:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Funnamed.webp?alt=media&token=72862fa3-6567-4043-aab0-cc449b30e602',
  },
  {
    id: '3',
    authorName: 'Dianne Russell',
    job: 'Blogger',
    quote:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Purus lorem id penatibus impedient. Turpis egestas ultricies purus  Lorem ipsum dolor sit amet2.',
    avatar:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fcute-avatar.webp?alt=media&token=7b6d918e-3dec-43cf-a4bd-70afdbdc63e0',
  },
]

export const OUR_MENU_MOCK = [
  PRICE_LIST_MOCK,
  PRICE_LIST_MOCK,
  PRICE_LIST_MOCK,
  PRICE_LIST_MOCK,
]

export const FEATURES_MOCK = [CARD_FEATURE, CARD_FEATURE, CARD_FEATURE]

export const BLOG_SECTION_MOCK = [CARD_BLOG, CARD_BLOG]

export const CATEGORY_SECTION_MOCK = [
  CARD_CATEGORY,
  CARD_CATEGORY,
  CARD_CATEGORY,
]
