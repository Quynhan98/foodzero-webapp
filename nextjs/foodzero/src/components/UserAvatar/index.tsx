import { memo } from 'react'
import { Avatar, Box, Flex, Heading, Text } from '@chakra-ui/react'

interface UserAvatarProps {
  src: string
  name: string
  job?: string
}

const UserAvatar = ({ src, name, job }: UserAvatarProps) => {
  return (
    <Flex alignItems="center" gap={{ base: '30px', md: '60px' }}>
      <Avatar
        name={name}
        src={src}
        width={{ base: '80px', md: '170px' }}
        height={{ base: '80px', md: '170px' }}
      />
      <Box>
        <Heading
          as="h5"
          variant="secondary"
          size={{ base: 'extraSmall', md: 'small' }}
        >
          {name}
        </Heading>
        <Text size={{ base: 'small', md: 'base' }}>{job}</Text>
      </Box>
    </Flex>
  )
}

export default memo(UserAvatar)
