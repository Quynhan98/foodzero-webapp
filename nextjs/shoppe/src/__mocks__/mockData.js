export const PRODUCT_ITEM = {
  id: 1,
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/test-1-50371.appspot.com/o/shoppe-products%2FImg-04.png?alt=media&token=641c3cfe-cab6-4498-8632-eb511434e042',
  name: 'Lira Earrings',
  price: 20,
  status: 'Sold out',
}
