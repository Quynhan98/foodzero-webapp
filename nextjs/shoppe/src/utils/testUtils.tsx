import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>{component}</ChakraProvider>,
  )
}

export * from '@testing-library/react'
export { customRender as render }
