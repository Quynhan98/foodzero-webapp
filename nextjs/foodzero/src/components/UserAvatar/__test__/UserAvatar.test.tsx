import { render, screen } from '@utils/testUtils'

// Mocks
import { AVATAR_MOCK } from '@mocks/mockData'

// Components
import UserAvatar from '@components/UserAvatar'

describe('UserAvatar component', () => {
  const props = {
    src: AVATAR_MOCK,
    name: 'John Doe',
    job: 'Bloger',
  }
  it('Renders UserAvatar correctly', () => {
    const { container } = render(<UserAvatar {...props} />)

    expect(container).toMatchSnapshot()
  })

  it('Should render data correctly', () => {
    render(<UserAvatar {...props} />)

    const name = screen.getByRole('heading')
    const job = screen.getByText(/bloger/i)

    expect(name).toHaveTextContent(props.name)
    expect(job).toHaveTextContent(props.job)
  })
})
