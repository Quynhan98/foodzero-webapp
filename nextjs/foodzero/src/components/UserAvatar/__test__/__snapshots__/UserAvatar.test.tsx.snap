// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar component Renders UserAvatar correctly 1`] = `
<div>
  <div
    class="css-mgvg8t"
  >
    <span
      class="chakra-avatar css-1kva3cr"
    >
      <div
        aria-label="John Doe"
        class="chakra-avatar__initials css-uo84k3"
        role="img"
      >
        JD
      </div>
    </span>
    <div
      class="css-0"
    >
      <h5
        class="chakra-heading css-gixbyc"
      >
        John Doe
      </h5>
      <p
        class="chakra-text css-lv0zmh"
      >
        Bloger
      </p>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
